experimental/packages/opentelemetry-instrumentation/src/platform/node/instrumentation.ts

```diff
@@ -116,18 +116,23 @@ export abstract class InstrumentationBase<T = any>
           }
         }
       }
-    } else {
-      // internal file
-      const files = module.files ?? [];
-      const file = files.find(f => f.name === name);
-      if (file && isSupported(file.supportedVersions, version, module.includePrerelease)) {
-        file.moduleExports = exports;
-        if (this._enabled) {
-          return file.patch(exports, module.moduleVersion);
-        }
-      }
     }
-    return exports;
+    // internal file
+    const files = module.files ?? [];
+    const matchingFileInstrumentations = files.filter(f => f.name === name);
+    // will return unmodified `exports` if the matching list is empty
+    return matchingFileInstrumentations.reduce<T>(
```

Comment:
not sure why you are using reduce since you don't mutate the exports, i would prefer to have the same behavior as before

Reply:
The original behaviour was returning the result of the `file.patch` call if there was a match, so I'm trying to do the same with multiple matches with reduce.
Should I just patch the same exports object and then return it?

Reply:
Indeed i missed that, i guess i would prefer to filter patches then apply them twice and return the exports object WDYT ?

Reply:
Something like this?
```diff
diff --git a/experimental/packages/opentelemetry-instrumentation/src/platform/node/instrumentation.ts b/experimental/packages/opentelemetry-instrumentation/src/platform/node/instrumentation.ts
index 7d8e8498..41766f68 100644
--- a/experimental/packages/opentelemetry-instrumentation/src/platform/node/instrumentation.ts
+++ b/experimental/packages/opentelemetry-instrumentation/src/platform/node/instrumentation.ts
@@ -119,20 +119,16 @@ export abstract class InstrumentationBase<T = any>
     }
     // internal file
     const files = module.files ?? [];
-    const matchingFileInstrumentations = files.filter(f => f.name === name);
-    // will return unmodified `exports` if the matching list is empty
-    return matchingFileInstrumentations.reduce<T>(
-      (patchedExports, file) => {
-        if (isSupported(file.supportedVersions, version, module.includePrerelease)) {
-          file.moduleExports = patchedExports;
-          if (this._enabled) {
-            return file.patch(patchedExports, module.moduleVersion);
-          }
-        }
-        return patchedExports;
-      },
-      exports,
-    );
+    const supportedFileInstrumentations = files
+      .filter(f => f.name === name)
+      .filter(f => isSupported(f.supportedVersions, version, module.includePrerelease));
+    supportedFileInstrumentations.forEach((file) => {
+      file.moduleExports = exports;
+      if (this._enabled) {
+        file.patch(exports, module.moduleVersion);
+      }
+    });
+    return exports;
   }
 
   public enable(): void {
```